docs/platforms/apple/guides/ios/session-replay/customredact.mdx

```diff
@@ -0,0 +1,66 @@
+---
+title: Using custom redact for Session Replay
+sidebar_order: 5501
+notSupported:
+description: "Learn how to choose which part of your app's data to redact in Session Replay."
+---
+
+<Note>
+
+Using custom redaction for Session Replay can expose sensitive data. Make sure to double-check every part of your app's data that you choose to redact or not.
+
+</Note>
+
+By default, our Session Replay SDK masks all text content, images, and user input, giving you heightened confidence that no sensitive data will leave the device. However, you can choose which parts of your app's data to redact or not by using some different options.
+
+
+## Redact by View class
+
+You can choose which type of view you want to redact or ignore by using the `redactViewClasses` or `ignoreViewClasses` options.
+
+Let's say you have a custom view that you want to redact and a UILabel subclass (which normally would be redacted) that you want to ignore. You can set the options like this:
+
+```swift
+  options.experimental.sessionReplay.redactViewClasses = [MyCustomView.self]
+  options.experimental.sessionReplay.ignoreViewClasses = [MyCustomLabel.self]
+```
+
+## Redact View by instance
+
+You can also choose to redact or ignore a specific view instance by using the replay API or view extensions like this:
+
+```swift
+  SentrySDK.replay.redactView(view: view)
+  SentrySDK.replay.ignoreView(view: label)
+```
+or
+
+```swift
+  view.sentryReplayRedact()
+  label.sentryReplayIgnore()
+```
+
+## SwiftUI
+
+Because of the way SwiftUI is transformed into UIKit, it will be often over-redacted. Modifier like `background` use the same element as an `Image`.
+In order to be able to control SwiftUI redact process, first you need to disable the default redaction options:
```

Comment:
(nitpick on grammar)
- "In order to **be able to control** SwiftUI redact process..." -> "In order to control **the** SwiftUI redact process..."

Reply:
Also (speaking as a total novice) it is not 100% clear if I need to disable the default options to be able to ignore view instances as described above (my intuition would say no, so even with default redaction I can manually un-redact views)

Reply:
```suggestion
In order to control the SwiftUI redact process, you first need to disable the default redaction options:
```